definitions/features/hammer.rb

```diff
@@ -92,7 +92,8 @@ def on_invalid_host(custom_config)
   end
 
   def on_invalid_password(custom_config)
-    if !ready? && custom_config[:foreman][:password].nil?
+    username = custom_config[:foreman][:username]
+    if !ready? && custom_config[:foreman][:password].nil? && (username == username_from_answers)
       msg = 'Invalid admin password was found in hammer configs. Looking into installer answers'
       logger.info(msg)
       custom_config[:foreman][:password] = password_from_answers
```

Comment:
We should test the username in here instead but Id suggest to test the initial_admin_user is the admin we account we need to use:
`if username_from_answers == feature(:foreman_server).admin_username`